packages/block-library/src/site-logo/edit.js

```diff
@@ -416,10 +416,12 @@ export default function LogoEdit( {
 		} );
 	};
 
-	const setIcon = ( newValue ) =>
+	const setIcon = ( newValue ) => {
+		newValue = newValue === undefined ? null : newValue;
 		editEntityRecord( 'root', 'site', undefined, {
 			site_icon: newValue,
```

Comment:
```suggestion
			site_icon: newValue ?? null,
```

Suggestion: We could use `nullish coalescing` here instead of re-assigning the value at the top.